### PR TITLE
Add tests for checking the minimum supported version to CI

### PR DESCRIPTION
**Pull request by UebelAndre**
This adds new builds to CI to test the root of the repo as well as the examples. Among those new builds are two checks for the minimum supported Bazel version.

In the future it might be better to consolidate all these tests into smaller tests cases such that we only have jobs for testing the root, examples, and docs directories. But I think making sure everything is being tested is a better place to be in.

### CODE REVIEW

File: .bazelci/config.yaml [hunk 1]
```diff
@@ -1,69 +1,102 @@
...
     platform: windows
     working_directory: examples
     test_flags:
       - "--enable_runfiles"
     test_targets:
       - "//:win_tests"
-  ubuntu1804_flags:
+  ubuntu2004_flags:
+    name: Flags
     platform: ubuntu1804
```
Comment (jsharpe) [inline .bazelci/config.yaml:32]: Needs updating to 2004

File: .bazelci/config.yaml [hunk 2]
```diff
@@ -1,69 +1,102 @@
...
     working_directory: test/standard_cxx_flags_test
     test_targets:
       - "//:flags_test"
-  ubuntu1604_flags:
+  ubuntu1804_flags:
+    name: Flags
     platform: ubuntu1604
```
Comment (jsharpe) [inline .bazelci/config.yaml:37]: Needs update to 1804

File: .bazelci/config.yaml [hunk 3]
```diff
@@ -1,69 +1,102 @@
...
   windows_flags:
+    name: Flags
     platform: windows
     working_directory: test/standard_cxx_flags_test
     test_targets:
       - "//:flags_test"
-  ubuntu1804_detect_root:
+  ubuntu2004_detect_root:
+    name: Detect root
     platform: ubuntu1804
```
Comment (jsharpe) [inline .bazelci/config.yaml:52]: Need updating to 2004

File: .bazelci/config.yaml [hunk 4]
```diff
@@ -1,69 +1,102 @@
...
     working_directory: test/detect_root_test
     test_targets:
       - "//:tests"
-  ubuntu1604_detect_root:
+  ubuntu1804_detect_root:
+    name: Detect root
     platform: ubuntu1604
```
Comment (jsharpe) [inline .bazelci/config.yaml:57]: Needs updating to 1804

File: .bazelci/config.yaml [hunk 5]
```diff
@@ -73,4 +106,21 @@ tasks:
     platform: ubuntu2004
     working_directory: docs
     build_targets:
-    - //...
\ No newline at end of file
+    - //...
+  min_supported_version:
+    name: "Minimum Supported Version"
+    bazel: "3.4.0"
+    platform: ubuntu1604
```
Comment (jsharpe) [inline .bazelci/config.yaml:113]: I assume this is 1604 because this is the only place where 3.4.0 is available in CI?

**Review by jsharpe — CHANGES_REQUESTED** [review]

File: .bazelci/config.yaml (hunk 5 above)
Comment (UebelAndre) [reply .bazelci/config.yaml:113]: I don't think so. I didn't want to drop support for 1604 here so I use it for the min version tests. If we decide we don't aren't going to test on that environment, I can easily change it but wanted that to be a well communicated decision.

**Comment by UebelAndre** [comment]
Thanks for the review, updated

**Review by jsharpe — APPROVED** [review]
